cfme/fixtures/automate.py

```diff
@@ -34,3 +34,32 @@ def klass(namespace):
     )
     yield klass
     klass.delete_if_exists()
+
+
+@pytest.fixture(scope="module")
+def request_cls(domain):
+    """custom request class"""
+    original_class = (
+        domain.parent.instantiate(name="ManageIQ")
+        .namespaces.instantiate(name="System")
+        .classes.instantiate(name="Request")
+    )
+    original_class.copy_to(domain=domain)
+    yield domain.namespaces.instantiate(name="System").classes.instantiate(name="Request")
+
+
+@pytest.fixture(scope="module")
+def custom_instance(request_cls):
+    """Custom instance; Need to pass ruby method code for this fixture"""
+    def method(ruby_code):
+        meth = request_cls.methods.create(
+            name="meth_{}".format(fauxfactory.gen_alphanumeric(4)), script=ruby_code
+        )
+
+        instance = request_cls.instances.create(
+            name="inst_{}".format(fauxfactory.gen_alphanumeric(4)),
+            fields={"meth1": {"value": meth.name}},
+        )
+        return instance
```

Comment:
Here as well.. I cant see where are these things being deleted.

Reply:
It will be part of the `domain`. so as `domain` deleted at teardown it will automatically clear. No need separate finalizer as per me.